### ISSUE TITLE
Discussion topic - Foreign key issues with reparent children

### ISSUE THREAD

**Issue by allenwq** [Coursemology/coursemology2#1228]
When trying to delete a course:

```
ActiveRecord::InvalidForeignKey (PG::ForeignKeyViolation: ERROR:  insert or update on table "course_discussion_posts" violates foreign key constraint "fk_course_discussion_posts_parent_id"
DETAIL:  Key (parent_id)=(11260) is not present in table "course_discussion_posts".
: UPDATE "course_discussion_posts" SET "parent_id" = 11260 WHERE "course_discussion_posts"."id" IN (SELECT "course_discussion_posts"."id" FROM "course_discussion_posts" WHERE "course_discussion_posts"."parent_id" = $1  ORDER BY "course_discussion_posts"."created_at" ASC)):
  app/models/course/discussion/post.rb:105:in `reparent_children'
  lib/extensions/destroy_callbacks/active_record/base.rb:28:in `update_status'
  lib/extensions/destroy_callbacks/active_record/base.rb:28:in `update_status'
  lib/extensions/destroy_callbacks/active_record/base.rb:28:in `update_status'
  lib/extensions/destroy_callbacks/active_record/base.rb:28:in `update_status'
  lib/extensions/destroy_callbacks/active_record/base.rb:28:in `update_status'
  app/controllers/system/admin/instance/courses_controller.rb:13:in `destroy'
  app/controllers/concerns/application_user_time_zone_concern.rb:13:in `set_time_zone'
```

**Comment by kxmbrian** [comment]
Check if topic is being destroyed. If so, do not re-parent.

**Comment by kxmbrian** [comment]
Fixed in #1845
